apps/medon-fe/src/components/RegistrationForm/index.tsx

```diff
@@ -0,0 +1,333 @@
+import { Controller, useForm } from 'react-hook-form';
+import { Input } from 'antd';
+import dayjs from 'dayjs';
+import { NavLink } from 'react-router-dom';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useTranslation } from 'react-i18next';
+import { registrationFormSchema } from 'validation/registrationFormSchema';
+import countries from 'utils/countries.json';
+import timezones from 'utils/timezones.json';
+import { ROLES } from 'utils/constants/roles';
+import { DATE_FORMAT_REG } from 'utils/constants/dateFormat';
+import {
+  BackBtn,
+  Btn,
+  BtnContainer,
+  Container,
+  ErrorMsg,
+  Form,
+  InputContainer,
+  Label,
+  LabelShort,
+  LabelText,
+  PassErrorMsg,
+  PasswordContainer,
+  StyledDatePicker,
+  StyledSelect,
+} from './styles';
+import { FormData } from './types';
+
+export default function RegistrationForm() {
+  const { t } = useTranslation();
+  const {
+    control,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<FormData>({
+    resolver: yupResolver(registrationFormSchema),
+    defaultValues: {
+      firstName: '',
+      lastName: '',
+      role: null,
+      speciality: null,
+      birthday: null,
+      country: null,
+      city: '',
+      timezone: '(UTC) Coordinated Universal Time',
+    },
+  });
+
+  const role = watch('role');
```

Comment:
![Uploading Screenshot 2023-04-10 at 17.40.03.png…]()

Reply:
changed to constant